automation/test-rhel8.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+_oc() { 
+  cluster/kubectl.sh "$@"
+}
+
+template_name="rhel8"
+
+# Prepare PV and PVC for rhel8 testing
+
+_oc create -f - <<EOF
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: disk-rhel
+  labels:
+    kubevirt.io/test: "rhel"
+spec:
+  capacity:
+    storage: 30Gi
+  accessModes:
+    - ReadWriteOnce
+  nfs:
+    server: "nfs"
+    path: /
+  storageClassName: rhel
+---
+EOF
+
+sleep 10
+
+_oc create -f - <<EOF
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: disk-rhel
+  labels:
+    kubevirt.io: ""
+spec:
+  volumeName: disk-rhel
+  storageClassName: rhel
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 30Gi
+
+  selector:
+    matchLabels:
+      kubevirt.io/test: "rhel"
+---
+EOF
+
+timeout=600
+sample=30
+
+kubeconfig="cluster/$KUBEVIRT_PROVIDER/.kubeconfig"
+
+sizes=("tiny" "small" "medium" "large")
+workloads=("desktop" "server" "highperformance")
+for size in ${sizes[@]}; do
+  for workload in ${workloads[@]}; do
+    templatePath="../../dist/templates/$template_name-$workload-$size.yaml"
+
+    _oc process -o json --local -f $templatePath NAME="$template_name-$workload-$size" PVCNAME=disk-rhel | \
+    jq '.items[0].spec.template.spec.volumes[0]+= {"ephemeral": {"persistentVolumeClaim": {"claimName": "disk-rhel"}}} | 
+    del(.items[0].spec.template.spec.volumes[0].persistentVolumeClaim)' | \
+    _oc apply -f -
+
+    # start vm
+    ./virtctl --kubeconfig=$kubeconfig start "$template_name-$workload-$size"
+
+    set +e
+    current_time=0
+    while [[ $(_oc get vmi $template_name-$workload-$size -o json | jq '.status.phase') != *Running* ]] ; do 
+      _oc describe vmi $template_name-$workload-$size
+      current_time=$((current_time + sample))
+      if [ $current_time -gt $timeout ]; then
+        exit 1
+      fi
+      sleep $sample;
+    done
+    set -e
+
+
+    # get ip address of vm
+    ipAddressVMI=$(_oc get vmi $template_name-$workload-$size -o yaml | grep ipAddress | awk '{print $3}')
+
+    set +e
+    current_time=0
+    # Make sure vm is ready
+    while _oc exec -it winrmcli -- ping -c1 $ipAddressVMI| grep "Destination Host Unreachable" ; do 
```

Comment:
do we need winrmcli for RHEL8 tests?

Reply:
this container is needed to check if VM is online or not (it pings pod with VM). Of course we can use e.g. busybox or something similar.

Reply:
I am fine with this, but it looks weird to see winrmcli in rhel8 test :) I understand we are using it just because it already includes the necessary tools.